apis/externalsecrets/v1/secretstore_yandexcertificatemanager_types.go

```diff
@@ -40,4 +40,12 @@ type YandexCertificateManagerProvider struct {
 	// The provider for the CA bundle to use to validate Yandex.Cloud server certificate.
 	// +optional
 	CAProvider *YandexCertificateManagerCAProvider `json:"caProvider,omitempty"`
+
+	// FetchByID configures the provider to interpret the `data.secretKey.remoteRef.key` field in ExternalSecret as certificate ID
+	// +optional
+	FetchByID *FetchByID `json:"fetchByID,omitempty"`
```

Comment:
What's the reason for even adding this if there is literally no check for this being not nil if `FetchByName` is not nil? I would just say, fetching by ID is the default otherwise if FetchByName is set, use that. There is no need for this value.

Reply:
In that case make this a policy setting that defines fecthing rules. We have similar things elsewhere. Basically make it an enum.

Reply:
> make this a policy setting

Could you please give some example? Do you mean something like https://external-secrets.io/latest/api/spec/#external-secrets.io/v1.ExternalSecretCreationPolicy?

The problem with making “fetching policy” an enum is that each value has its own (potentially extending) set of parameters:
-   “fetching by ID”: no parameters
-   “fetching by name”: `folderID`
-   there is also a possibility that at least “fetching by labels” will be added in the future with a parameter like `labelQuery`

Our first thought was to make “fetching policy” enum, but later we decided to use [SecretStoreProvider approach](https://github.com/external-secrets/external-secrets/blob/v0.19.2/apis/externalsecrets/v1/secretstore_types.go#L67-L211) as a reference.
Was it a bad idea?

Reply:
Thanks!

Reply:
thanks, I'll review it when I have the time. :)

Reply:
@Skarlso gentle ping :)